.github/workflows/deploy.yml

```
@@ -16,12 +16,13 @@ on:
-# Cancel a queued deploy if a newer commit lands while one is running.
-# Cloudflare Pages handles a fast cancel gracefully (the in-flight one
-# completes; the new one queues right after).
+# One serialized production publication lane exists. Queueing every candidate
+# prevents an obsolete historical rerun from canceling or replacing an eligible
+# current-main deployment before the freshness resolver can classify it. Stale
+# candidates reach that resolver in turn and become provider-free no-ops.
 concurrency:
-  group: deploy-${{ github.ref }}
-  cancel-in-progress: false
+  group: deploy-production-main
+  queue: max
```

actionlint (1.7.12)

[error] unexpected key "queue" for "concurrency" section. expected one of "cancel-in-progress", "group"

(syntax-check)